.github/actions/do_build_dpcpp/action.yml

```diff
@@ -0,0 +1,122 @@
+name: build dpc++
+description: build dpc++
+
+inputs:
+  target:
+    description: 'target architecture'
+  download_dpcpp_artefact:
+    description: 'none | dpcpp_release' # TODO: add support for 'previous workflow' download
+    type: string
+    default: "none"
+
+runs:
+  using: "composite"
+  steps:
+    - name: calc vars
+      id: calc_vars
+      uses: ./.github/actions/calc_vars
+      with:
+        target: ${{ inputs.target }}  
+
+    - name: Install Ninja
+      uses: llvm/actions/install-ninja@main
+
+    - name: Install spirv tools
+      shell: bash
+      run:    
+        sudo apt-get install spirv-tools
+    
+    - name: clone dpc++
+      if: inputs.download_dpcpp_artefact == 'none'
+      uses: actions/checkout@v4
+      with:
+        repository: intel/llvm
+        path: llvm
+
+    - name: dpcpp configure
+      if: inputs.download_dpcpp_artefact == 'none'
+      shell: bash
+      run:
+        cd llvm; python buildbot/configure.py
+        -o build
+        --host-target="X86;AArch64;RISCV"
+        --native_cpu
+        --llvm-external-projects=lld
+        --cmake-opt=-DNATIVECPU_USE_OCK=ON
+        --cmake-opt=-DLLVM_ENABLE_ZLIB=OFF
+        --cmake-opt=-DLLVM_ENABLE_ZSTD=OFF
+
+    - name: build sycl-headers
+      if: inputs.download_dpcpp_artefact == 'none'
```

Comment:
I've gone with create_ock_artefact_method == 'build/<workflow id>' I think it would make sense to follow that, since it gives more flexibility. However if you want to do it in a further PR that's fine.

Reply:
Going with:
```
 create_dpcpp_artefact_method:
    description: 'build | download_release' # TODO: add support for 'previous workflow' download
```

Reply:
Done